legal-api/src/legal_api/reports/report.py

```diff
@@ -586,8 +587,36 @@ def _format_consent_continuation_out_data(self, filing):
             self._format_address(filing['offices']['registeredOffice']['mailingAddress'])
 
     def _format_agm_extension_data(self, filing):
-        # FUTURE: format logic for letter of agm extension
-        return
+        meta_data = self._filing.meta_data or {}
+        is_first_agm = meta_data.get('agmExtension', {}).get('isFirstAgm', '')
+        filing['is_first_agm'] = is_first_agm
+        filing['agm_year'] = meta_data.get('agmExtension', {}).get('year', '')
+        filing['is_final_agm'] = meta_data.get('agmExtension', {}).get('isFinalExtension', '')
+
+        number_world = ['one', 'two', 'three', 'four', 'five', 'six']
+        duration_numeric = meta_data.get('agmExtension', {}).get('extensionDuration', '')
+        filing['duration_numeric'] = duration_numeric
+        filing['duration_spelling'] = number_world[int(duration_numeric)-1]
+
+        if is_first_agm:
+            funding_date_json = self._filing.filing_json['filing'].get('business', {}).get('foundingDate', '')
+            funding_date = funding_date_json[0:10]
+            original_date_time = LegislationDatetime.\
+                as_legislation_timezone_from_date_str(funding_date) + relativedelta(months=18)
+            filing['original_agm_date'] = original_date_time.strftime(OUTPUT_DATE_FORMAT)
+        else:
+            expire_date_current_string = meta_data.get('agmExtension', {}).get('expireDateCurrExt', '')
+            date_current_obj = datetime.strptime(expire_date_current_string, '%Y-%m-%d')
```

Comment:
Maybe you can also use method at https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/utils/legislation_datetime.py#L58 to get a date time object. 👀